drm/savage/Makefile

```diff
@@ -3,7 +3,7 @@
 .PATH:	${.CURDIR:H:H}/src/dev/drm
 
 KMOD= 	savage
-CFLAGS += -I${SRCTOP}/sys/dev/drm
+CFLAGS += -I${.CURDIR:H}/sys/dev/drm
```

Comment:
I'm not sure this is right. What are you trying to accomplish?

Reply:
Just as @johalun said, SRCTOP is not defined, so make ends up looking in /sys/dev/drm. This directory is not available on head.

Reply:
I still don't see how this makes the build fail. I have no problem building this in poudriere on the latest HEAD, where sys/dev/drm is missing.

I am not sure this is needed at all, to be honest. I could build drm-legacy-kmod even though this is removed.

Reply:
Here's the error I get without any patches:
```
===> drm/savage (all)
machine -> /usr/src/sys/powerpc/include
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/device_if.m -h
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/bus_if.m -h
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/dev/pci/pci_if.m -h
:> opt_drm.h
Warning: Object directory not changed from original /tmp/usr/ports/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/drm/savage
cc  -O2 -pipe -fno-strict-aliasing  -Werror -D_KERNEL -DKLD_MODULE -nostdinc  -I/sys/dev/drm -I/tmp/usr/ports/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/drm/../src/ -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common  -mlongcall -fno-omit-frame-pointer     -MD  -MF.depend.savage_bci.o -MTsavage_bci.o -mno-altivec -msoft-float -ffreestanding -fwrapv -fstack-protector -Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -fformat-extensions -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wno-uninitialized  -finline-limit=15000 -fms-extensions --param inline-unit-growth=100 --param large-function-growth=1000 -mcall-aixdesc  -std=iso9899:1999 -c /tmp/usr/ports/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/src/dev/drm/savage_bci.c -o savage_bci.o
cc1: error: /sys/dev/drm: No such file or directory
*** Error code 1
```

Reply:
How are you building the port?  Can you try on amd64 as well?
I'm very curious why this is only an issue for you on ppc64.

Reply:
On amd64, I can only try building on 12.0-RELEASE, which still has drm in base.

On this, I can see:
```
===> drm/savage (all)
machine -> /usr/src/sys/amd64/include
x86 -> /usr/src/sys/x86/include
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/device_if.m -h
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/bus_if.m -h
awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/dev/pci/pci_if.m -h
:> opt_drm.h
Warning: Object directory not changed from original /usr/local/poudriere/ports/default/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/drm/savage
cc  -O2 -pipe -fno-strict-aliasing  -Werror -D_KERNEL -DKLD_MODULE -nostdinc  -I/sys/dev/drm -I/usr/local/poudriere/ports/default/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/drm/../src/ -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common  -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer   -MD  -MF.depend.savage_bci.o -MTsavage_bci.o -mcmodel=kernel -mno-red-zone -mno-mmx -mno-sse -msoft-float  -fno-asynchronous-unwind-tables -ffreestanding -fwrapv -fstack-protector -Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wno-error-tautological-compare -Wno-error-empty-body -Wno-error-parentheses-equality -Wno-error-unused-function -Wno-error-pointer-sign -Wno-error-shift-negative-value -Wno-address-of-packed-member  -mno-aes -mno-avx  -std=iso9899:1999 -c /usr/local/poudriere/ports/default/graphics/drm-legacy-kmod/work/drm-legacy-3c8c990/src/dev/drm/savage_bci.c -o savage_bci.o
```

It builds, but you can see `-I/sys/dev/drm` in there.

Reply:
How are you building the port? from /usr/ports or similar, or with poudriere, or some other way?

Reply:
/usr/ports, but same can be seen with poudriere.

Reply:
Looking at my build logs. I can see `-I/sys/dev/drm`, but I can't see the error. Is ppc64 built with gcc or clang?

Can you test with removing this CFLAG completely?

Reply:
ppc64 uses gcc 4.2.

Building this with CFLAGS removed from both Makefiles also passes.

Reply:
It's probably a difference between gcc and clang in how it handles missing -I then, that makes it so we haven't detected the issue on amd64 and i386 (which both use clang).  Can you update the patch and simply remove the include altogether?

Reply:
Done